aea/multiplexer.py

```diff
@@ -335,9 +342,14 @@ async def disconnect(self) -> None:
                 self.connection_status.set(ConnectionStates.disconnecting)
                 await asyncio.wait_for(self._stop(), timeout=60)
                 self.logger.debug("Multiplexer disconnected.")
-            except (CancelledError, Exception):
+            except (CancelledError) as e:
+                self.logger.exception("Exception on disconnect:")
+                raise AEAConnectionError("Failed to disconnect the multiplexer.") from e
+            except Exception as e:
                 self.logger.exception("Exception on disconnect:")
-                raise AEAConnectionError("Failed to disconnect the multiplexer.")
+                raise AEAConnectionError(
+                    f"Failed to disconnect the multiplexer cause {repr(e)}"
```

Comment:
I would use the same error message format as before, i.e.
```suggestion
                    f"Failed to disconnect the multiplexer: Error: {repr(e)}"
```